lib/Target/Lattigo/LattigoEmitter.cpp

```diff
@@ -318,9 +316,23 @@ LogicalResult LattigoEmitter::printOperation(arith::ConstantOp op) {
 }
 
 LogicalResult LattigoEmitter::printOperation(tensor::ExtractOp op) {
-  // only support 1-dim tensor for now
   os << getName(op.getResult()) << " := " << getName(op.getTensor()) << "[";
-  os << getName(op.getIndices()[0]) << "]\n";
+  os << flattenIndexExpression(
+      op.getTensor().getType(), op.getIndices(),
+      [&](Value value) { return variableNames->getNameForValue(value); });
```

Comment:
Use the `getName` wrapper for `variableNames->getNameForValue(value);`. Same below.